### PR TITLE
Handling mapped type Partial

### PR DESCRIPTION
**Pull request by benbabic**
Guys, 

This PR provides a test case for ts type `Partial` and an implementation that passes the test without breaking the existing ones.
I did not update the README.md as I did not know what to put in there but I'm more than happy to do it if you want.

Not really sure it works for all mapped types but it works for `Partial`

Please read the commit messages for technical details.

Relates to #95

### CODE REVIEW

File: test/programs/partial-mapped-type/main.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+type MyObject = Partial<{
+    required: number;
+    optional: number;
```
Comment (domoritz) [inline test/programs/partial-mapped-type/main.ts]: It's a bit confusing that both variables are optional but one has the name `required`. Why not only one?

Comment (benbabic) [inline test/programs/partial-mapped-type/main.ts]: Indeed, I just lazily copy / pasted the optional test. Will change that to something more meaningful later today. Any suggestions for the names ?

**Comment by domoritz** [comment]
Can you create a nested object with one object that is partial and one that
is not? I'd like to test that partial actually works and not also pass if
required is disabled.

On Sat, Mar 18, 2017, 13:24 Benjamin Babic <notifications@github.com> wrote:

*@benbabic* commented on this pull request.
------------------------------

In test/programs/partial-mapped-type/main.ts
<https://github.com/YousefED/typescript-json-schema/pull/105#discussion_r106786196>
:

> @@ -0,0 +1,4 @@
+type MyObject = Partial<{
+    required: number;
+    optional: number;

Indeed, I just lazily copy / pasted the optional test. Will change that to
something more meaningful later today. Any suggestions for the names ?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/YousefED/typescript-json-schema/pull/105#discussion_r106786196>,
or mute the thread

**Comment by domoritz** [comment]
Any new on this @benbabic?

**Comment by domoritz** [comment]
Closing because of inactivity.

**Pull request closed without merging**